ogip/lc.py

```diff
@@ -0,0 +1,238 @@
+
+    ###
+
+double_time_definition_keywords = [ ('MJDREFI', 'MJDREFF'), 
+                                    ('TSTARTI', 'TSTARTF'),
+                                    ('TSTOPI', 'TSTOPF'),
+                                    ('TIMEZERI', 'TIMEZERF'),
+                                    ]
+
+single_time_definition_keywords = ['MJDREF',
+                                    'TSTART',
+                                    'TSTOP',
+                                    'TIMEZERO',
+                                    'TIMEUNIT',
+                                   'TIMEDEL',
+                                   ]
+
+optional_keywords = ['TELAPSE',
+                    'TIMEREF',                     
+                     'TIMESYS', 
+                     'CLOCKCOR', 
+                     'DATE-OBS' , 
+                     'DATE-END',
+                    'TELESCOP', 
+                    'INSTRUME', 
+                    'FILTER',
+                    'OBJECT',
+                    'RA',
+                    'DEC',
+                    'EQUINOX',
+                    'RADECSYS',
+                    'ORIGIN',
+                    'DATE',
+                    'AUTHOR'
+                    'CREATOR']
+
+
+class rate(LightCurve):
+    _time = None
+    _timedel = None
+    _fracexp = None
+    _keywords = {}
+
+    _filename = None
+
+    def __init__(self, filename):
+
+        self._keywords = {kk : 0.0 for kk in single_time_definition_keywords}
+
+        methods_tried= {}
+        issue = True
+        for n, m in rate.__dict__.items():
+            
+            if n.startswith('from_file_name_'):
+                try:
+                    self.__getattribute__(m.__name__)(filename)
+                    issue = False
+                except Exception as e:
+                    logger.debug("failed to read with %s  %s", m.__name__,  e)
+                    methods_tried[n] = e
+        if issue:
+            raise Exception(f"failed to read with any method, tried: {methods_tried}")
+
+
+    def read_keywords(self, hdu):
+         
+        for kk in single_time_definition_keywords + optional_keywords:
+            #print(kk)
+            if kk in hdu.header:
+                self._keywords[kk] = hdu.header[kk]
+                logger.debug(f'Updating {kk}')
+        for cc in double_time_definition_keywords:
+            #print(cc[0], cc[1])
+            if cc[0] in hdu.header and cc[1] in hdu.header:
+                tmp_val = hdu.header[cc[0]] + hdu.header[cc[1]]
+                for kk in single_time_definition_keywords[0:4]:
+                    if cc[0].startswith(kk[0:-1]):
+                        logger.debug(f'Updating {kk} from {cc[0]} and {cc[1]}')
+                        self._keywords[kk] = tmp_val
+        
+        
+    
+    def from_file_name_osa(self, fn):
+        f = fits.open(fn)
+        issue = True
+        for e in f:
+            try:                   
+                    self._filename = fn
```

Comment:
this is double tab. Also there are other style anomalies, and to settle on a standard I added a test action, please rebase with that.

Reply:
I have run the tester locally and it reveals no more issues.